agent/grpc-external/services/peerstream/stream_resources.go

```diff
@@ -301,8 +301,15 @@ func (s *Server) HandleStream(streamReq HandleStreamRequest) error {
 			}
 
 			if resp := msg.GetResponse(); resp != nil {
+
+				st, found := s.Tracker.MutableStreamStatus(streamReq.LocalID)
```

Comment:
We already have the status from back on line 168 so there is no need to look it up again.

Reply:
oh, I missed that 🤦🏼